tests/unit/api/test_delegations.py

```diff
@@ -0,0 +1,228 @@
+# SPDX-FileCopyrightText: 2024 Repository Service for TUF Contributors
+#
+# SPDX-License-Identifier: MIT
+
+import json
+from datetime import timezone
+
+import pretend
+from fastapi import status
+
+DELEGATIONS_URL = "/api/v1/delegations/"
+DELEGATIONS_DELETE_URL = "/api/v1/delegations/delete"
+MOCK_PATH = "repository_service_tuf_api.delegations"
+
+
+class TestPostDelegationAPI:
+    def test_post_delegation(self, test_client, monkeypatch, fake_datetime):
+        """Test creating a new delegation via POST /api/v1/delegations/"""
+        # Mock metadata_delegation function
+        mocked_delegation_response = pretend.stub(
+            data=pretend.stub(
+                task_id="fake_task_id",
+                last_update=fake_datetime.now(timezone.utc),
+            ),
+            message="Metadata delegation add accepted.",
+        )
+        mocked_metadata_delegation = pretend.call_recorder(
+            lambda payload, action: mocked_delegation_response
+        )
+        monkeypatch.setattr(
+            f"{MOCK_PATH}.metadata_delegation", mocked_metadata_delegation
+        )
```

Comment:
Can we mock instead the `metadata_delegation`, mock one level below, so we can test the implementation through the API, adding a good coverage and going beyond, looking for changes in that implementation.

In that way we will let the `metadata_delegation` function run :) 

Mocking for example:
`get_task_id()`
`metadata_delegation.apply_async()`

See my example below with the HTTPException